src/components/page-filters/page-filters.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { Button, KIND, SIZE } from 'baseui/button';
+import { Search, Filter, Delete } from 'baseui/icon';
+import { Input } from 'baseui/input';
+
+import { Props } from './page-filters.types';
+import { styled, overrides } from './page-filters.styles';
+import {
+  PageQueryParamKeys,
+  PageQueryParamSetterValues,
+  PageQueryParams,
+} from '@/hooks/use-page-query-params/use-page-query-params.types';
+import usePageQueryParams from '@/hooks/use-page-query-params/use-page-query-params';
+
+export default function PageFilters<T extends PageQueryParams>({
+  search,
+  searchId,
+  setSearch,
+  searchPlaceholder,
+  pageFiltersConfig,
+  pageQueryParamsConfig,
+}: Props<T>) {
+  const [areFiltersShown, setAreFiltersShown] = React.useState(false);
+  const [queryParams, setQueryParams] = usePageQueryParams(
+    pageQueryParamsConfig,
+    { pageRerender: false }
+  );
+
+  const activeFiltersCount = React.useMemo(
+    () =>
+      pageFiltersConfig.filter((pageFilter) =>
+        pageFilter.isSet({ queryParams })
+      ).length,
+    [pageFiltersConfig, queryParams]
+  );
+
+  const resetAllFilters = React.useCallback(() => {
+    setQueryParams(
+      pageQueryParamsConfig.reduce(
+        (acc: Partial<PageQueryParamSetterValues<T>>, config) => {
+          const queryParamKey: PageQueryParamKeys<T> = config.key;
+          if (queryParamKey !== searchId) {
```

Comment:
When we discussed this, you'd mentioned that in all our use cases, the page query params are all related to filters (except search). The only way to clear these filters otherwise would be to pass an array of query params to clear.

Reply:
Seems to be a miscommunication. The page can have other queryParams.
In the configuration we have for the domain page search is the only param that shouldn't be cleared, but this is not the general case.